aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OnSuccessConsumer.kt

```diff
@@ -0,0 +1,28 @@
+package com.amplifyframework.datastore.syncengine
```

Comment:
On the subject of Kotlin vs Java, we need to be thoughtful in which one we choose.  For this `OnSuccessConsumer` (and `OnErrorConsumer`), since it is so closely tied to the `SyncProcessor`, I think we should be using the same language (Java).  One reason for this is that it makes refactoring difficult.  For example, if we wanted to move a chunk of code from `SyncProcessor` to `OnSuccessConsumer`, or vice versa, we can no longer just copy and paste, like we could if they were written in the same language, so it will likely slow us down a bit.